File: app/src/main/java/com/apptest/Users/BaseUserFragment.java
package com.apptest.Users;

import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ProgressBar;
import android.widget.TextView;

import com.apptest.R;
import com.apptest.adapter.UsersAdapter;
import com.apptest.models.User;

import butterknife.BindView;

/**
 * Created by davidpayel on 06/02/2017.
 */

public class BaseUserFragment extends Fragment {
    @BindView(R.id.recyclerview) protected RecyclerView recyclerview;
    @BindView(R.id.text_info) protected TextView textInfo;
    @BindView(R.id.progress) protected ProgressBar progress;

    protected void initRecyclerView() {
        UsersAdapter adapter = new UsersAdapter();
        adapter.onItemClickListener(new UsersAdapter.OnItemClickListener() {
            @Override
            public void onItemClick(User user, View view) {

            }
        });
        recyclerview.setAdapter(adapter);
        recyclerview.setLayoutManager(new LinearLayoutManager(getActivity()));
    }
}

File: app/src/main/java/com/apptest/Users/base/BaseUserFragment.kt
package com.apptest.Users.base

import android.support.annotation.Nullable
import android.support.design.widget.Snackbar
import android.support.v4.app.Fragment
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.CheckBox
import android.widget.ProgressBar
import android.widget.TextView

import com.apptest.R
import com.apptest.Users.adapters.UsersAdapter
import com.apptest.Users.models.User

import butterknife.BindView

/**
 * Created by davidpayel on 06/02/2017.
 */

open class BaseUserFragment : Fragment() {
    @Nullable @JvmField @BindView(R.id.recyclerview) var recyclerview: RecyclerView? = null
    @Nullable @JvmField @BindView(R.id.text_info) protected var textInfo: TextView? = null
    @Nullable @JvmField @BindView(R.id.progress) protected var progress: ProgressBar? = null

    protected fun initRecyclerView() {
        val adapter = UsersAdapter(activity, object : UsersAdapter.OnItemClickListener {
            override fun onItemClick(user: User, v: View) {
                Snackbar.make(v, user.name.toString(), Snackbar.LENGTH_LONG).show()
            }
        })
        recyclerview?.adapter = adapter
        recyclerview?.layoutManager = LinearLayoutManager(activity)
    }
}
